includes/API/ContextAuthorized.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace DataAccounting\API;
+
+use MediaWiki\Permissions\PermissionManager;
+use MediaWiki\Rest\HttpException;
+use MediaWiki\Rest\SimpleHandler;
+use LogicException;
+use RequestContext;
+
+abstract class ContextAuthorized extends SimpleHandler {
+
+	/**
+	 * @var PermissionManager 
+	 */
+	protected $permissionManager;
+
+	/**
+	 * @param PermissionManager $permissionManager
+	 */
+	public function __construct( PermissionManager $permissionManager ) {
+		$this->permissionManager = $permissionManager;
+	}
+
+	public function execute() {
+		$paramSettings = $this->getParamSettings();
+		$validatedParams = $this->getValidatedParams();
+		$unvalidatedParams = [];
+		$params = [];
+		foreach ( $this->getRequest()->getPathParams() as $name => $value ) {
+			$source = $paramSettings[$name][self::PARAM_SOURCE] ?? 'unknown';
+			if ( $source !== 'path' ) {
+				$unvalidatedParams[] = $name;
+				$params[] = $value;
+			} else {
+				$params[] = $validatedParams[$name];
+			}
+		}
+
+		if ( $unvalidatedParams ) {
+			throw new LogicException(
+				'Path parameters were not validated: ' . implode( ', ', $unvalidatedParams )
+			);
+		}
+
+		$this->checkPermission( ...$params );
+
+		// @phan-suppress-next-line PhanUndeclaredMethod
+		return $this->run( ...$params );
+	}
+
+	/** @inheritDoc */
+	public function needsWriteAccess() {
+		return false;
+	}
+
+	/** @inheritDoc */
+	public function needsReadAccess() {
+		return false;
+	}
+
+	/**
+	 * @return string[]
+	 */
+	protected function getPermissions(): array {
+		return [ 'read' ];
+	}
+
+	protected function checkPermission() {
+		// @phan-suppress-next-line PhanUndeclaredMethod
+		$title = $this->provideTitle( ...func_get_args() );
```

Comment:
@pwirth does this mean `title` is now a required URL param of the read REST API requests?

Reply:
How does this guard against someone requesting the rev id details of a protected page, while specifying the title of a whitelisted page?

Reply:
the title is not a parameter. the rest module extending this abstract class is required to generate a title from the already given params. this would either be the revision id or the page title

Reply:
OK make sense.